src/serialization/RDB/decoder/current/v1/decode_v1.c

```diff
@@ -0,0 +1,260 @@
+#include "decode_v1.h"
+#include "assert.h"
+
+/**
+ * In case of IO errors, the default return values are:
+ * numbers - 0
+ * strings - null
+ * So only when it is necessary check for IO errors.
+ */
+
+void *RAI_RDBLoadTensor_v1(RedisModuleIO *io) {
+    int64_t *shape = NULL;
+    int64_t *strides = NULL;
+
+    DLContext ctx;
+    ctx.device_type = RedisModule_LoadUnsigned(io);
+    ctx.device_id = RedisModule_LoadUnsigned(io);
+    if (RedisModule_IsIOError(io))
+        goto cleanup;
+
+    // For now we only support CPU tensors (except during model and script run)
+    assert(ctx.device_type == kDLCPU);
+    assert(ctx.device_id == 0);
+
+    DLDataType dtype;
+    dtype.bits = RedisModule_LoadUnsigned(io);
+    dtype.code = RedisModule_LoadUnsigned(io);
+    dtype.lanes = RedisModule_LoadUnsigned(io);
+
+    size_t ndims = RedisModule_LoadUnsigned(io);
+    if (RedisModule_IsIOError(io))
+        goto cleanup;
+
+    shape = RedisModule_Calloc(ndims, sizeof(*shape));
+    for (size_t i = 0; i < ndims; ++i) {
+        shape[i] = RedisModule_LoadUnsigned(io);
+    }
+
+    strides = RedisModule_Calloc(ndims, sizeof(*strides));
+    for (size_t i = 0; i < ndims; ++i) {
+        strides[i] = RedisModule_LoadUnsigned(io);
+    }
+
+    size_t byte_offset = RedisModule_LoadUnsigned(io);
+
+    size_t len;
+    char *data = RedisModule_LoadStringBuffer(io, &len);
+    if (RedisModule_IsIOError(io))
+        goto cleanup;
+
+    RAI_Tensor *ret = RedisModule_Calloc(1, sizeof(*ret));
+    ret->tensor = (DLManagedTensor){.dl_tensor = (DLTensor){.ctx = ctx,
+                                                            .data = data,
+                                                            .ndim = ndims,
+                                                            .dtype = dtype,
+                                                            .shape = shape,
+                                                            .strides = strides,
+                                                            .byte_offset = 0},
+                                    .manager_ctx = NULL,
```

Comment:
Why the byte_offset field is initialized to 0 and not to the loaded byte_offset?